metrics/scaling/net-serve.json.in

```diff
@@ -0,0 +1,40 @@
+{
+    "apiVersion": "apps/v1",
+    "kind": "Deployment",
+    "metadata": {
+        "labels": {
+            "run": "net-serve"
+        },
+        "name": "@DEPLOYMENT@"
+    },
+    "spec": {
+        "replicas": 1,
+        "selector": {
+            "matchLabels": {
+                "run": "net-serve"
+            }
+        },
+        "template": {
+            "metadata": {
+                "labels": {
+                    "run": "net-serve",
+                    "@LABEL@": "@LABELVALUE@"
+                }
+            },
+            "spec": {
+                "terminationGracePeriodSeconds": @GRACE@,
+                "runtimeClassName": "@RUNTIMECLASS@",
+                "automountServiceAccountToken": false,
+                "containers": [{
+                    "name": "net-serve",
+                    "image": "nginx",
```

Comment:
I'm not wholly against `nginx`, but:
- I think nginx is larger than the e2e test team agnhost image
- a k8s e2e image exists, so I thought it better we use that
- we've seen problems in the past where nginx serves the test page up off the backing storage, every time, which caused us a performance measure issue before - I'm not sure if that is the same for agnhost or not though.

You can see how to use the agnhost for an http liveness test over at https://github.com/kata-containers/tests/blob/master/integration/kubernetes/runtimeclass_workloads/pod-http-liveness.yaml and https://github.com/kata-containers/tests/blob/master/integration/kubernetes/k8s-liveness-probes.bats#L35-L50 (although I'm not sure the latter actually does an http check ...).

Reply:
OK, if you can't make `agnhost` work (having a quick look, something like its [netexec](https://github.com/kubernetes/kubernetes/tree/master/test/images/agnhost#netexec) `echo` mode looked appropriate to me), then let's roll with nginx - just keep one eye out for any oddities it might introduce if we end up doing throughput tests etc.

Reply:
Since there is a notable position to use `agnhost` image, I've invested more time to get the `agnhost` image work on this proposed test and **big kudos** to [@ericho](https://github.com/ericho) who helped me to get this done with the proper net test using `nc` 🎉